source/yats/include/yats/scheduler.h

```diff
@@ -17,7 +18,12 @@ class scheduler
 {
 public:
     explicit scheduler(const pipeline& pipeline)
-        : m_tasks(pipeline.build())
+        : scheduler(pipeline, 4)
```

Comment:
when you have optional arguments it is usually the best to put the as default arguments and not as its own constructors. The reason for this is that number_of_threads gets an litteral value assigned that is not visible by looking at the interface.